components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2477,59 +2478,81 @@ subroutine ma_resuspend_convproc(                           &
 !-----------------------------------------------------------------------
 ! arguments
 ! (note:  TMR = tracer mixing ratio)
-   integer,  intent(in)    :: pcnst_extd
+   integer,  intent(in)    :: pcnst_extd ! cloudborne aerosol dimension [unitless]
    real(r8), intent(inout) :: dcondt(pcnst_extd,pver)
-                              ! overall TMR tendency from convection
+                              ! overall TMR tendency from convection [kg/kg/s] 
    real(r8), intent(inout) :: dcondt_resusp(pcnst_extd,pver)
-                              ! portion of TMR tendency due to resuspension
+                              ! portion of TMR tendency due to resuspension [kg/kg/s]
                               ! (actually, due to the adjustments made here)
-   real(r8), intent(in)    :: const(pcnst_extd,pver)  ! TMRs before convection
 
-   real(r8), intent(in)    :: dp_i(pver) ! pressure thickness of level (in mb)
+   real(r8), intent(in)    :: dp_i(pver) ! pressure thickness of level [mb]
    integer,  intent(in)    :: ktop, kbot_prevap ! indices of top and bottom cloud levels ! REASTER 08/05/2015
 
 !-----------------------------------------------------------------------
 ! local variables
    integer  :: k, ll, la, lc, n
-   real(r8) :: qa, qc, qac           ! working variables (mixing ratios)
    real(r8) :: qdota, qdotc, qdotac  ! working variables (MR tendencies)
 !-----------------------------------------------------------------------
 
 
    do n = 1, ntot_amode
 
-      do ll = 0, nspec_amode(n)
-         if (ll == 0) then
-            la = numptr_amode(n)
-            lc = numptrcw_amode(n) + pcnst
-         else
-            la = lmassptr_amode(ll,n)
-            lc = lmassptrcw_amode(ll,n) + pcnst
-         end if
+      ! TMR number
+      la = numptr_amode(n)
+      lc = numptrcw_amode(n) + pcnst
+      if ((la > 0) .and. (la <= pcnst_extd) .and. (lc > 0) .and. (lc <= pcnst_extd)) then
```

Comment:
Since you are sending la, lc and pcnst_extd as arguments, we can move this check inside the tmr_tendency:
```
if ((la > 0) .and. (la <= pcnst_extd) .and. (lc > 0) .and. (lc <= pcnst_extd)) then
     do k = ktop, kbot_prevap ! REASTER 08/05/2015
            qdota = dcondt(la,k)
            qdotc = dcondt(lc,k)
            qdotac = qdota + qdotc
            ! cam5 approach
            dcondt(la,k) = qdotac
            dcondt(lc,k) = 0.0
            dcondt_resusp(la,k) = (dcondt(la,k) - qdota)
            dcondt_resusp(lc,k) = (dcondt(lc,k) - qdotc)
   enddo
endif